src/server/course/course.service.ts

```diff
@@ -30,4 +30,24 @@ export class CourseService {
       })),
     });
   }
+
+  /**
+   * Resolve an array containing all catalog prefixes that currently exist in the
+   * database, as strings
+   */
+  public async getCatalogPrefixList(): Promise<string[]> {
+    return this.courseRepository
+      .createQueryBuilder('c')
+      .select('c.prefix', 'prefix')
+      .distinct(true)
```

Comment:
One issue here is that in the four-year plan interface where we'll be using this data, the courses in the list are only the SEAS/EPS courses, as defined in the [MultiYearPlanView entity](https://github.com/seas-computing/course-planner/blob/2cc517d570680cb5cf8f33610d38c545e3a0fcf9/src/server/courseInstance/MultiYearPlanView.entity.ts#L21). If this part of the metadata is only going to be used in the four-year plan and the schedule (which also only shows SEAS courses) it probably makes sense to limit this query to only courses where `"isSEAS" <> IS_SEAS.N`.

Reply:
Oh, I see, ok. I updated the course service in 3636a96 along with two course service test setups to account for the new `where` clause.